GUI-pause-menu/Assets/Scripts/PauseMenu.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UIElements;
+using Unity.UIElements.Runtime;
+
+namespace GUI
+{
+    public class PauseMenu : MonoBehaviour
+    {
+        /// <summary>
+        ///     Uxml/Uss ui element class selectors.
+        ///     Match strings with Uxml class names and types.
+        ///     
+        ///     C_MENU_BUTTON type = Button
+        ///     C_CLOSE_BUTTON type = Button
+        ///     C_SCREEN_CONTAINER type = VisualElement
+        ///     TEMPLATE_CONTAINER type = TemplateContainer (embedded uxml)
+        ///     DEFAULT_SCREEN = VisualElement
+        /// </summary>
+        public const string C_MENU_BUTTON = "menu-button";              
+        public const string C_CLOSE_BUTTON = "close-button";
+        public const string C_SCREEN_CONTAINER = "screen-container";
+        public const string TEMPLATE_CONTAINER = "screen-template";
+        public const string DEFAULT_SCREEN = "screen-I";
+
+        public Button closeButton;
+
+        public VisualElement currentScreen;
+
+        public List<Button> menuButtons = new List<Button>();
+
+        public List<VisualElement> screenContainers = new List<VisualElement>();
+
+        // Store screen templates for targeting 
+        public List<TemplateContainer> screens = new List<TemplateContainer>();
+
+        // Panel renderer (display UI in runtime scene) 
+        private PanelRenderer pr;
+
+        // Virtual Tree (root) for querys
+        private VisualElement vt;
+
```

Comment:
Use of abbreviations (pr, vt) is debatable. Not explicitly specified in the NoolWok's Unity Style Guide though..

Reply:
Ook met Frank besproken, `pr` ga ik renamen naar `panelRenderer` en `vt` naar `root`. `vt` is netzoals jquery een soort selector (`$`) binnen de virtual tree a.k.a. `root`